software/contrib/pong.py

```diff
@@ -0,0 +1,230 @@
+"""
+Pong
+author: Igor Nawrocki (github.com/Bugari)
+date: 2023-05-19
+labels: game
+
+EuroPi version of the classic Pong game.
+
+digital_in: unused
+analog_in: unused
+
+knob_1: controls position of the left paddle
+knob_2: controls position of the right paddle
+
+button_1: start a new game
+button_2: unused
+
+output_1: trigger a sound for the ball hitting a wall
+output_2: trigger a sound for the ball hitting a paddle
+output_3: trigger a sound for scoring
+output_4: unused
+output_5: unused
+output_6: unused
+
+"""
+
+import math
+import time
+import random
+from time import sleep
+from europi import oled, b1, k1, k2, Output, cv1, cv2, cv3
+from europi_script import EuroPiScript
+
+
+class Paddle:
+    LEFT = -1
+    RIGHT = 1
+    NONE = 0
+
+class SoundTrigger:
+    def __init__(self, cv: Output):
+        self.trigger_time_ms = 50
+        self.cv = cv
+        self.triggered = False
+        self.last_trigger_time = 0
+
+
+    def trigger(self):
+        self.cv.on()
+        self.triggered = True
+        self.last_trigger_time = time.ticks_ms()
+
+    def update(self):
+        if self.triggered:
+            if time.ticks_ms() - self.last_trigger_time > self.trigger_time_ms:
+                self.cv.off()
+                self.triggered = False
+
+    def reset(self):
+        self.cv.off()
+        self.triggered = False
+
+
+wall_sound = SoundTrigger(cv1)
+paddle_sound = SoundTrigger(cv2)
+score_sound = SoundTrigger(cv3)
+
+
+sounds = [wall_sound, paddle_sound, score_sound]
+
+class Ball:
```

Comment:
[Discussion] I find the responsibilities of the two main classes here (`Ball` and `Pong`) hard to follow they seem very coupled. For example, it seems like both classes have some responsibilities for the `SoundTriggers`. It may simplify things to narrow the scope of the ball class to only the physics and drawing of the ball itself, and move the game related code into the pong class.

As a way to clarify this, consider a hypothetical case where you wanted to add the ability to play the game with two balls.